src/providers/mssql/qgsmssqlprovider.cpp

```diff
@@ -417,7 +417,7 @@ void QgsMssqlProvider::loadFields()
         }
         if ( sqlType == QVariant::String )
         {
-          int length = query.value( 7 ).toInt();
+          int length = query.value( ( sqlTypeName == "uniqueidentifier" ) ? 6 : 7 ).toInt();
```

Comment:
Can you make this a `QLatin1String` and add an explanation why 6 or 7 are appropriate?